### PR TITLE
Add/ load config from filesystem

### PR DESCRIPTION
**Pull request by rgadon107**

### CODE REVIEW

**Comment by rgadon107** [comment]
@hellofromtonya Hi Tonya, This branch is ready for code review. 

While I've indirectly tested the function `_load_config_from_filesystem( $path_to_file )`, I still don't know if it's possible to test it directly with a string value added in for $path_to_file. 

If there any more assertions I should test for, please let me know.

File: mu-plugins/central-hub/src/config-store/internals.php [hunk 1]
```diff
@@ -69,6 +69,11 @@ function _the_store( $store_key = '', $config_to_store = array(), $remove = null
  * @return array
  */
 function _load_config_from_filesystem( $path_to_file ) {
+
+	if ( ! $path_to_file ) {
```
Comment (hellofromtonya) [inline mu-plugins/central-hub/src/config-store/internals.php]: @rgadon107 This conditional isn't necessary.  Why?  The `$path_to_file` is a required parameter.  When you invoke the function, you have to pass an argument to it.  If the value received is a falsey, such as `false`, `null`, etc., an error will be thrown when the `require` runs.

File: mu-plugins/central-hub/tests/phpunit/unit/config-store/_loadConfigFromFilesystem.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	protected function setUp() {
+		parent::setUp();
+
+		require_once CENTRAL_HUB_ROOT_DIR . '/src/config-store/internals.php';
+	}
+
+	/**
+	 *  Test should return false when $path_to_file is empty string.
+	 */
+	public function test_should_return_false_when_path_to_file_is_empty_string() {
```
Comment (hellofromtonya) [inline mu-plugins/central-hub/tests/phpunit/unit/config-store/_loadConfigFromFilesystem.php]: I don't think this is a valid edge case.  Why would anyone pass an empty string to this function?  

But if they did, it would throw an error.  Instead, test that an error is thrown here when a falsey is passed.

Hint:  It'll also throw an error if the path/to/file doesn't exist.

File: mu-plugins/central-hub/tests/phpunit/unit/config-store/_loadConfigFromFilesystem.php [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		$expected = false;
+
+		$this->assertFalse( _load_config_from_filesystem( '') );
+		$this->assertSame( $expected, _load_config_from_filesystem( '' ) );
+	}
+
+	/**
+	 *  Test should return array key and current array when a configuration array is given.
+	 */
+	public function test_should_return_array_key_and_current_array_from_config() {
```
Comment (hellofromtonya) [inline mu-plugins/central-hub/tests/phpunit/unit/config-store/_loadConfigFromFilesystem.php]: What is this test doing?  It's not testing the `_load_config_from_filesystem` function.

Here, you want to test that the configuration array is returned when you specify a valid path/to/file.  Right?  

Here's what to do:

1. In `tests/phpunit/` create a folder called `fixtures`.
2. Add a dummy configuration file and then add the configuration parameters to it.
3. Then in this test:
      - give `_load_config_from_filesystem` the path/to/config file you just created in the `fixtures` folder
      - test that it sends back the array in the right format, i.e. as expected

File: mu-plugins/central-hub/src/config-store/internals.php (hunk 1 above)
Comment (rgadon107) [reply mu-plugins/central-hub/src/config-store/internals.php]: @hellofromtonya This conditional was removed on 05-10-19 following the discussion in our code review.

File: mu-plugins/central-hub/tests/phpunit/unit/config-store/_loadConfigFromFilesystem.php (hunk 2 above)
Comment (rgadon107) [reply mu-plugins/central-hub/tests/phpunit/unit/config-store/_loadConfigFromFilesystem.php]: @hellofromtonya Following our code review, this test was removed.

File: mu-plugins/central-hub/tests/phpunit/unit/config-store/_loadConfigFromFilesystem.php (hunk 3 above)
Comment (rgadon107) [reply mu-plugins/central-hub/tests/phpunit/unit/config-store/_loadConfigFromFilesystem.php]: @hellofromtonya [1] A `fixtures` directory was added to `/tests/phpunit`; [2] a dummy configuration array was added; and [3] the 'path/to/file' was assigned to the variable `$path_to_file` within an updated test for the function `_load_config_from_filesystem`.  

A new test to evaluate that the array fixture was loaded was written and all assertions passed.

**Comment by rgadon107** [comment]
@hellofromtonya Hi! I responded to your comments, and updated the branch. If there's nothing further to test in that function, I'd like to merge the `add/_loadConfigFromFilesystem` branch into the `development` branch.

File: mu-plugins/central-hub/tests/phpunit/fixtures/test-cpt-config.php [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+ */
+
+namespace spiralWebDb\centralHub\Tests\Fixtures;
+
+return  [
+	'foo' => [
+		'aaa' => 'bbb',
+		'ccc' => 'ddd'
+	]
+];
```
Comment (hellofromtonya) [inline mu-plugins/central-hub/tests/phpunit/fixtures/test-cpt-config.php]: @rgadon107 Make sure there's an empty line at the end of PHP files.

File: mu-plugins/central-hub/tests/phpunit/unit/config-store/_loadConfigFromFilesystem.php [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	}
+
+	/**
+	 *  Test should return array when $path_to_file is given.
+	 */
+	public function test_should_return_array_when_path_to_file_is_given() {
+		$path_to_file = CENTRAL_HUB_ROOT_DIR . '/tests/phpunit/fixtures/test-cpt-config.php';
+		$config       = _load_config_from_filesystem( $path_to_file );
+
+		$this->assertArrayHasKey( 0, $config );
```
Comment (hellofromtonya) [inline mu-plugins/central-hub/tests/phpunit/unit/config-store/_loadConfigFromFilesystem.php]: @rgadon107 You want to test against the known fixture configuration to ensure what you get back from `_load_config_from_filesystem` is exactly what you expect.

Here's the code in that function:

https://github.com/rgadon107/cornerstone/blob/7de92a8b0c1a54c3de12a6846b903ba852a272a2/mu-plugins/central-hub/src/config-store/internals.php#L71-L78

Here's what you want to test:

```php
$path_to_file = CENTRAL_HUB_ROOT_DIR . '/tests/phpunit/fixtures/test-cpt-config.php';
$expected = [
     'aaa' => 'bbb',
     'ccc' => 'ddd'
];
$actual       = _load_config_from_filesystem( $path_to_file );

$this->assert( 'foo', $actual[0] );
$this->assertSame( $expected, $actual[1] );
```

1. Test that index 0 is the key.
2. Test that index 1 is the expected configuration parameters array.

File: mu-plugins/central-hub/tests/phpunit/fixtures/test-cpt-config.php (hunk 4 above)
Comment (rgadon107) [reply mu-plugins/central-hub/tests/phpunit/fixtures/test-cpt-config.php]: Added new line at end of file.

File: mu-plugins/central-hub/tests/phpunit/unit/config-store/_loadConfigFromFilesystem.php (hunk 5 above)
Comment (rgadon107) [reply mu-plugins/central-hub/tests/phpunit/unit/config-store/_loadConfigFromFilesystem.php]: Tests updated per code review.

File: mu-plugins/central-hub/tests/phpunit/fixtures/test-cpt-config.php (hunk 4 above)
Comment (rgadon107) [reply mu-plugins/central-hub/tests/phpunit/fixtures/test-cpt-config.php]: Added new line at end of file per code review.
